pkg/router/headerformatter.go

```diff
@@ -47,6 +66,73 @@ func (f *plainHeaderFormatter) append() bool {
 	return f.isAppend
 }
 
-func (f *plainHeaderFormatter) format(requestInfo types.RequestInfo) string {
+func (f *plainHeaderFormatter) format(_ types.HeaderMap, _ types.RequestInfo) string {
 	return f.staticValue
 }
+
+type variableHeaderFormatter struct {
+	isAppend     bool
+	variableName string
+}
+
+func (v *variableHeaderFormatter) format(headers types.HeaderMap, requestInfo api.RequestInfo) string {
+	var (
+		prefix   string
+		isPrefix bool
+	)
+	if strings.HasPrefix(v.variableName, types.VarPrefixReqHeader) {
+		prefix = types.VarPrefixReqHeader
+		isPrefix = true
+	} else if strings.HasPrefix(v.variableName, types.VarPrefixRespHeader) {
+		prefix = types.VarPrefixRespHeader
+		isPrefix = true
+	}
+	if isPrefix {
+		headerName := strings.TrimPrefix(v.variableName, prefix)
```

Comment:
maybe we should move these lines to `getHeaderFormatter`? for better performance.

Reply:
yeah, I will add a new headerformatter to deal with this situation.